code/game/machinery/spaceheater.dm

```diff
@@ -186,9 +186,11 @@
 	if(!can_interact(user))
 		return
 	on = !on
+	mode = HEATER_MODE_STANDBY
+	user.visible_message(span_notice("[user] switches [on ? "on" : "off"] \the [src]."), span_notice("You switch [on ? "on" : "off"] \the [src]."))
 	update_appearance()
 	if (on)
-		START_PROCESSING(SSmachines, src)
+		SSair.start_processing_machine(src)
```

Comment:
This is all the same as the logic in ui_act right? could we move it to a separate proc and just call that instead of having two implementations of the same thing?

Would prevent errors like this in the future too

Reply:
Yes, it is!  I have moved it to a separate proc which is now called by both the right click action and the tgui functionality.  Good catch!